pkg/storegateway/series_refs.go

```diff
@@ -869,23 +877,19 @@ func fetchCachedSeriesForPostings(ctx context.Context, userID string, indexCache
 	return res, true
 }
 
-func storeCachedSeriesForPostings(ctx context.Context, indexCache indexcache.IndexCache, userID string, blockID ulid.ULID, matchers []*labels.Matcher, shard *sharding.ShardSelector, postingsKey indexcache.PostingsKey, set seriesChunkRefsSet, logger log.Logger) {
-	nonNilShard := maybeNilShard(shard)
-	matchersKey := indexcache.CanonicalLabelMatchersKey(matchers)
-	data, err := encodeCachedSeriesForPostings(set, matchersKey, nonNilShard)
+func storeCachedSeriesForPostings(ctx context.Context, indexCache indexcache.IndexCache, userID string, blockID ulid.ULID, shard *sharding.ShardSelector, itemID cachedSeriesForPostingsID, set seriesChunkRefsSet, logger log.Logger) {
+	data, err := encodeCachedSeriesForPostings(set, itemID.encodedPostings)
 	if err != nil {
-		logSeriesForPostingsCacheEvent(ctx, logger, userID, blockID, matchers, shard, postingsKey, "msg", "can't encode series for caching", "err", err)
+		logSeriesForPostingsCacheEvent(ctx, logger, userID, blockID, shard, itemID, "msg", "can't encode series for caching", "err", err)
 		return
 	}
-	indexCache.StoreSeriesForPostings(ctx, userID, blockID, matchersKey, shard, postingsKey, data)
+	indexCache.StoreSeriesForPostings(ctx, userID, blockID, shard, itemID.postingsKey, data)
 }
 
-func encodeCachedSeriesForPostings(set seriesChunkRefsSet, matchersKey indexcache.LabelMatchersKey, nonNilShard sharding.ShardSelector) ([]byte, error) {
+func encodeCachedSeriesForPostings(set seriesChunkRefsSet, postings []byte) ([]byte, error) {
```

Comment:
[nit] Just to clarify what's the expected input.

```suggestion
func encodeCachedSeriesForPostings(set seriesChunkRefsSet, encodedPostings []byte) ([]byte, error) {
```